libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationTelemetry.kt

```diff
@@ -109,12 +118,14 @@ internal object NavigationTelemetry : NavigationMetricListener {
         }
         updateLifecyclePercentages()
         // Send arrival event
-        NavigationMetricsWrapper.arriveEvent(
+        val event = NavigationEventFactory.buildNavigationArriveEvent(
+            PhoneState(context),
             navigationSessionState,
-            routeProgress,
+            MetricsRouteProgress(routeProgress),
             metricLocation.location,
-            context
+            sdkIdentifier ?: ""
```

Comment:
Actually, technically, `sdkIdentifier` can be null, in case when we call some method of `NavigationTelemetry` without calling `initialize` method. But I agree, as it `internal` class, this case is not valid. But we have tests that call some methods separately, without calling `initialize` method, and then this case become valid) So for testing purposes, I think, we need to leave it as is. WDYT?

Reply:
> But we have tests that call some methods separately, without calling initialize method, and then this case become valid) So for testing purposes, I think, we need to leave it as is.

Shouldn't we call `initialize` always to guarantee that `MapboxTelemetry` is created correctly even in tests?

Reply:
But this tests (that calls method with sdkIdentifier) not event linked at all with `NavigationTelemetry`. For example `onSuccess_currentLocationEngineCallbackUpdatesCorrectly()` test in `LocationUpdaterTest`

Reply:
I believe the problem is that we're coupling `NavigationTelemetry` in `LocationUpdater` 👀 https://github.com/mapbox/mapbox-navigation-android/blob/6b35dc7caa18d490c17002325cf8f84496b105bd/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/LocationUpdater.kt#L45

Let's leave it as is for now and revisit the tests and ☝️ in a future PR.